fvtest/porttest/omrsockTest.cpp

```diff
@@ -0,0 +1,124 @@
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "omrport.h"
+#include "omrportsock.h"
+#include "testHelpers.hpp"
+
+/**
+ * Start a server which creates a socket, binds to an address/port and 
+ * listens for clients.
+ * 
+ * @param[in] portLibrary
+ * @param[in] addrStr The address of the server.
+ * @param[in] port The server port.
+ * @param[in] family Socket address family wanted.
+ * @param[out] serverSocket A pointer to the server socket. 
+ * @param[out] serverAddr The socket address of the server created.
+ * 
+ * @return 0 on success, return an error otherwise.
+ */ 
+int32_t
+start_server(struct OMRPortLibrary *portLibrary, const char *addrStr, const char *port, int32_t family, omrsock_socket_t *serverSocket, omrsock_sockaddr_t serverAddr) 
+{
+    return OMRPORT_ERROR_NOTEXIST;
+}
+
+/**
+ * Create the client socket, and then connect to the server.
+ *  
+ * @param[in] portLibrary
+ * @param[in] addrStr The address of the server.
+ * @param[in] port The server port.
+ * @param[in] family Socket address family wanted.
+ * @param[in] sessionClientSocket A pointer to the client socket. 
+ * @param[in] sessionClientAddr The socket address of the client created. 
+ * 
+ * @return 0 on success, return an error otherwise.
+ */ 
+int32_t
+connect_client_to_server(struct OMRPortLibrary *portLibrary, const char *addrStr, const char *port, int32_t family, omrsock_socket_t *sessionClientSocket, omrsock_sockaddr_t sessionClientAddr) 
+{
+    return OMRPORT_ERROR_NOTEXIST;
+}
+
+/**
+ * Test if port library is properly set up to run OMRSock tests. 
+ *
+ * This will iterate through all OMRSock functions in @ref omrsock.c and simply check 
+ * that the location of the function is not NULL.
+ *
+ * @note Errors such as an OMRSock function call with a NULL address will be reported.
+ */
+TEST(PortSockTest, library_function_pointers_not_null)
+{
+    OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
```

Comment:
```
14:34:33  omrsockTest.cpp: In member function ‘virtual void PortSockTest_library_function_pointers_not_null_Test::TestBody()’:
14:34:33  omrsockTest.cpp:73:21: error: unused variable ‘privateOmrPortLibrary’ [-Werror=unused-variable]
14:34:33       OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
```
You can't have unused variables. You will have to remove this macro call.
```suggestion
    /* Unimplemented. */
```

Reply:
Thank you, I just fixed it. @rwy0717 Could you please run the tests again, thank you.